src/main/host/descriptor/tcp.c

```diff
@@ -822,7 +822,7 @@ static void _tcp_updateReceiveWindow(TCP* tcp) {
      * unordered input packets should count against buffer space, so use the _tcp version. */
     //gsize space = _tcp_getBufferSpaceIn(tcp); // causes throughput problems
     gsize space = legacysocket_getInputBufferSpace(&(tcp->super));
-    gsize nPackets = space / (CONFIG_MTU - CONFIG_HEADER_SIZE_TCPIP - CONFIG_HEADER_SIZE_ETH);
+    gsize nPackets = space / (CONFIG_MTU - CONFIG_HEADER_SIZE_TCPIP);
```

Comment:
Consider using the CONFIG_TCP_MAX_SEGMENT_SIZE constant defined in src/main/core/support/definitions.h.